### ISSUE TITLE
TypeError with Table

### ISSUE THREAD

**Issue by whottel** [ewels/MultiQC#1091]
**Description of bug:**
Get TypeError when sample names are heterogeneous when visualizing results in a table. For example two samples named `123` and `456-abc`.

**MultiQC Error log:**
```python
Module custom_content raised an exception: Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/multiqc-1.8.dev0-py3.6.egg/EGG-INFO/scripts/multiqc", line 440, in multiqc
    output = mod()
  File "/usr/lib/python3.6/site-packages/multiqc-1.8.dev0-py3.6.egg/multiqc/modules/custom_content/custom_content.py", line 226, in custom_module_classes
    parsed_modules.append( MultiqcModule(module_id, mod) )
  File "/usr/lib/python3.6/site-packages/multiqc-1.8.dev0-py3.6.egg/multiqc/modules/custom_content/custom_content.py", line 272, in __init__
    self.write_data_file( mod['data'], "multiqc_{}".format(modname.lower().replace(' ', '_')) )
  File "/usr/lib/python3.6/site-packages/multiqc-1.8.dev0-py3.6.egg/multiqc/modules/base_module.py", line 332, in write_data_file
    util_functions.write_data_file(data, fn, sort_cols, data_format)
  File "/usr/lib/python3.6/site-packages/multiqc-1.8.dev0-py3.6.egg/multiqc/utils/util_functions.py", line 78, in write_data_file
    for sn in sorted(data.keys()):
TypeError: '<' not supported between instances of 'str' and 'float'
```

**File that triggers the error:**
#contrived example, but should replicate error
[example.txt](https://github.com/ewels/MultiQC/files/4032806/example.txt)

**MultiQC run details (please complete the following):**
Using singularity for docker image of multiqc found here https://hub.docker.com/r/staphb/multiqc
Run on HPC cluster running CentOS Linux

***Additional note*** 
Issue seems to be unique to table as using other formats, such as bargraph does not produce this error with the same sample names.
Also error resolved if first name changed to `123-abc`

**Comment by ewels** [comment]
Thanks for reporting this @whottel - looks like I'm being a bit too clever with casting variable types in the code somewhere. Should be fixable by forcing strings for the sort step. I'll look into it when I get a chance 👍 

Phil

**Comment by ewels** [comment]
Fixed, will go out in v1.9 shortly. Thanks for reporting! Let me know if you spot any problems with it still.

Phil
